tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py

```diff
@@ -54,9 +50,49 @@
 from tensorflow.python.util import compat
 from tensorflow.python.util.tf_export import tf_export
 
+from tensorflow.python.keras.utils import tf_utils
+from tensorflow.python.distribute import distribution_strategy_context as distribute_ctx
+from tensorflow.python.distribute import values as distribute_values_lib
+from tensorflow.python.training.saving import saveable_object
+from tensorflow.python.distribute import distribute_utils
+
 _ANONYMOUS_TRAINABLE_STORE_KEY = '_anonymous_trainable_store_key'
 
 
+class TrainableWrapperDistributedPolicy(distribute_values_lib.OnWritePolicy):
+
+  def get_saveable(self, var, primary_var, name):
+    for v in var.values:
+      v._gather_saveables_for_checkpoint()
+
+    def tensor():
+      return array_ops.zeros((
+          0,
+          primary_var.params.dim,
+      ),
+                             dtype=primary_var.params.value_dtype)  # pylint: disable=protected-access
+
+    spec = saveable_object.SaveSpec(tensor=tensor,
+                                    slice_spec="",
+                                    name=name,
+                                    dtype=primary_var.params.value_dtype,
+                                    device=primary_var.device)
+    return tensor, [spec]
+
+
+class DistributedVariableWrapper(distribute_values_lib.DistributedVariable):
+
+  def __init__(self, strategy, values, aggregation, var_policy=None):
+    super(DistributedVariableWrapper, self).__init__(strategy, values,
+                                                     aggregation, var_policy)
+
+
+class DEResourceVariable(resource_variable_ops.ResourceVariable):
```

Comment:
What's the value of the class? It looks similar to the superclass?
Maybe you can try this if you only want to clean your code:
DEResourceVariable = resource_variable_ops.ResourceVariable`

Reply:
This allows TFRA to independently identify its own ResourceVariable for processing. Reference to:
```python
def trainable_wrapper_filter(iterable_object_in,
                             test_unaggregated_function=None):
```
https://github.com/MoFHeka/recommenders-addons/blob/cefa36a9daf37be129355b1bf70cb6f2e6143421/tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_ops.py#L1115

Maybe moving DistributedVariableWrapperFilter function to dynamic_embedding_ops.py would be better?